metaconfig-core/shared/src/main/scala/metaconfig/ConfDecoder.scala

```diff
@@ -41,6 +41,10 @@ trait ConfDecoder[A] { self =>
     NoTyposDecoder[A](self)
 }
 
+trait ConfDecoderFactory[A] {
+  def apply(default: A): ConfDecoder[A]
```

Comment:
Exact name up for debate, but can we come up with something else than `apply`?
```suggestion
  def decoder(default: A): ConfDecoder[A]
```

Reply:
done